.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,32 +1,49 @@
-
-* Enable the feature flag: `add_filter( 'sensei_feature_flag_{THE_FLAG_NAME}', '__return_true' );`
+Add as many details as possible to help others reproduce the issue and test the changes.
+"Before / After" screenshots can also be very helpful when the change is visual.
 -->
-*
 
+1.
+2.
+3.
+
+## New/Updated Hooks
 <!-- Add the following only if there are new/updated actions or filters. Please provide a brief description of what they do and any arguments they may take. Be sure to also add the "Hooks" label to this PR. -->
-### New/Updated Hooks
 
 *
 
+## Deprecated Code
 <!-- Add the following only if there is any code that is being deprecated. Please list the replacement function or hook that should be called instead, if applicable. Be sure to also add the "Deprecation" label to this PR. -->
-### Deprecated Code
 
 *
 
-<!--
-Helpful tips for screenshots:
-https://en.support.wordpress.com/make-a-screenshot/
--->
-### Screenshot / Video
+## Pre-Merge Checklist
+<!-- Complete applicable items on this checklist **before** merging. Items that are not applicable can be left unchecked.
+
+Both the PR author and reviewer are responsible for ensuring the checklist is completed. -->
+- [ ] PR title and description contain sufficient detail and accurately describe the changes
+- [ ] Acceptance criteria is met
+- [ ] Decisions are publicly documented
+- [ ] Adheres to coding standards ([PHP](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/), [JavaScript](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/), [CSS](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/), [HTML](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/html/))
+- [ ] All strings are translatable (without concatenation, handles plurals)
+- [ ] Data is [sanitized](https://developer.wordpress.org/apis/security/sanitizing/) / [escaped](https://developer.wordpress.org/apis/security/escaping/)
+- [ ] Follows our naming conventions (P6rkRX-4oA-p2)
+- [ ] Hooks (p6rkRX-1uS-p2) and functions are documented
+- [ ] New UIs are responsive and use a [mobile-first approach](https://zellwk.com/blog/how-to-write-mobile-first-css/)
+- [ ] New UIs match the designs
+- [ ] Test cases are written and passing (including feature flags)
+- [ ] Different user privileges (admin, teacher, subscriber) are tested as appropriate
+- [ ] Code is tested on the minimum supported PHP and WordPress versions
+- [ ] User interface changes have been tested on the latest versions of Chrome, Firefox and Safari
+- [ ] We are generally happy with it and don’t feel like it immediately needs to be rewritten
+- [ ] Application performance has not degraded
+- [ ] "Needs Documentation" label is added if this change requires updates to documentation
+- [ ] Known issues are created as new GitHub issues
+- [ ] Continuous Integration build is passing
+- [ ] Code is peer reviewed
```

Comment:
The last two checks are a bit redundant because of the project branch protection settings. Maybe we can shorten the list a bit?

Reply:
I'd like to leave the CI one for now. Given that the e2e and code coverage checks sometimes fail, and we are currently able to merge despite these failures, I want to ensure that people take the time to confirm that their PR doesn't introduce new failures.

I'll remove the code review one though.